lib/ConnectionManager.js

```diff
@@ -290,6 +300,14 @@ ConnectionManager.prototype.onSocketError = function (error) {
     // we will retry connect in that listener function.
 };
 
+ConnectionManager.prototype.onSessionTimeout = function () {
+    var connectStates = [STATES.CONNECTED, STATES.CONNECTED_READ_ONLY];
+
+    if (connectStates.indexOf(this.state) === -1) {
+        this.setState(STATES.SESSION_EXPIRED);
```

Comment:
I believe you also need to call `if(this.socket) this.socket.destroy()` after this to prevent a dangling socket.